File: py/solns/binaryTreePreorderTraversal/binaryTreePreorderTraversal.py
from solns.treeNode.treeNode import *
class Solution:
    @staticmethod
    def naive(root):
        res = []
        def preOrder(tree):
            if tree==None: return
            res.append(tree.val)
            preOrder(tree.left)
            preOrder(tree.right)
        preOrder(root)
        return res

File: cpp/example/src/BinaryTreePreorderTraversal/binaryTreePreorderTraversal.cpp
#include "binaryTreePreorderTraversal.h"

void binaryTreePreorderTraversal::recursive(TreeNode* tree){
    if (tree==nullptr) return;
    this->res.push_back(tree->val);
    recursive(tree->left);
    recursive(tree->right);
};

vector<int> binaryTreePreorderTraversal::naive(TreeNode* root){
    recursive(root);
    return this->res;    
};
